Sources/MonkeyKing/MonkeyKing+Error.swift

```diff
@@ -4,36 +4,44 @@ import UIKit
 extension MonkeyKing {
 
     public enum Error: Swift.Error {
-        case noAccount
-        case messageCanNotBeDelivered
-        case invalidImageData
-        case userCancelled
+
+        public enum ResourceReason {
+            case invalidImageData
+            case missingTitle
+            case missingDescription
+            case missingThumbnail
+            case missingMedia
+            case imageTooBig
+            case textTooLong
+        }
 
         public enum SDKReason {
             case invalidURLScheme
             case urlEncodeFailed
+            case urlDecodeFailed
             case serializeFailed
+            case deserializeFailed
             case other(code: String)
         }
 
-        case sdk(reason: SDKReason)
-
-        public struct APIRequestReason {
-            public enum `Type` {
-                case unrecognizedError
-                case connectFailed
-                case invalidToken
-            }
-
-            public var type: Type
-            public var responseData: [String: Any]?
+        public enum APIRequestReason {
+            case unrecognizedError(response: ResponseJSON?)
+            case connectFailed
+            case invalidToken
+            case invalidParameter
+            case missingParameter
         }
 
-        case apiRequest(reason: APIRequestReason)
+        case noApp
+        case noAccount
+        case userCancelled
+        case resource(ResourceReason)
+        case sdk(SDKReason)
+        case apiRequest(APIRequestReason)
     }
 
-    func errorReason(with responseData: [String: Any], at platform: SupportedPlatform) -> Error.APIRequestReason {
-        let unrecognizedReason = Error.APIRequestReason(type: .unrecognizedError, responseData: responseData)
+    func errorReason(with responseData: [String: Any], at platform: SupportedPlatform) -> Error {
```

Comment:
I prefer renaming the func name with `buildError(with:at:)` or others after you changed return type

Reply:
OK.